### PR TITLE
Default to Kokkos parallel backend

### PR DESCRIPTION
**Pull request by streeve**
By default, this will enable runtime backends in order as: HIP, Cuda, OpenMP, Serial (with commandline options to override). It also allows users to disable CabanaMD backends even if enabled in Kokkos.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -41,13 +41,9 @@ if( NOT Cabana_ENABLE_MPI )
...
 endif()
 
-option(CabanaMD_REQUIRE_SERIAL "Build CabanaMD with Serial support" ON)
-option(CabanaMD_REQUIRE_OPENMP "Build CabanaMD with OpenMP support" OFF)
-option(CabanaMD_REQUIRE_CUDA "Build CabanaMD with Cuda support" OFF)
-option(CabanaMD_REQUIRE_HIP "Build CabanaMD with HIP support" OFF)
-
 set(CABANAMD_SUPPORTED_DEVICES SERIAL OPENMP CUDA HIP)
 foreach(_device ${CABANAMD_SUPPORTED_DEVICES})
+  option(CabanaMD_REQUIRE_${_device} "Build CabanaMD with ${_device} support" ${Kokkos_ENABLE_${_device}})
```
Comment (dalg24) [inline CMakeLists.txt:46]: Why did you call it "require"?  It feels like it should be "enable".

**Review by dalg24 — COMMENTED** [review]
Why do you think it is worth the trouble of selectively enabling Kokkos backends rather than just take whatever is provided by the Kokkos that was found?

File: CMakeLists.txt (hunk 1 above)
Comment (streeve) [reply CMakeLists.txt:46]: I was just following Cabana

**Comment by streeve** [comment]
> Why do you think it is worth the trouble of selectively enabling Kokkos backends rather than just take whatever is provided by the Kokkos that was found?

Mainly trying to reduce compile times. But after discussion I agree that instead just enabling the default Kokkos space is all that's worthwhile

File: src/mdfactory.h [hunk 2]
```diff
@@ -24,15 +24,28 @@ class MDfactory
   public:
     static CabanaMD *create( InputCL commandline )
     {
+        // If device is not specifically enabled, choose in this order
         int device = commandline.device_type;
+        if ( device == NONE )
```
Comment (dalg24) [inline src/mdfactory.h:29]: I think this is still worth adding and picking up the kokkos default execution space

Comment (streeve) [reply src/mdfactory.h:29]: I opened #75 instead

**Pull request closed without merging**